h2o-core/src/main/java/water/parser/ParseSetup.java

```diff
@@ -633,6 +634,23 @@ private static final long maxLineLength(byte[] bytes) {
     return -1;
   }
 
+  /**
+   * Copies the setup to another object (that is possibly and extension of the base setup)
+   * @param setup target setup object
+   * @param <T> class derived from ParseSetup
+   * @return the target setup object (for convenience)
+   */
+  public <T extends ParseSetup> T copyTo(T setup) {
+    try {
+      for (Field field : ParseSetup.class.getDeclaredFields()) {
```

Comment:
I am little bit confused here:

Do you want only fields from `ParseSetup.class` OR all fields from `this` `this.getClass().getDeclaredFields() + inherited fields`. 

The current implementation will not work for `self <:< ParseSetup` properly, since it copies fields declared by  ParseSetup only. But then you have to redefine the method to be something like `<S extends ParseSetup, D extends S> copyTo(S source, D destination)`